charm.go

```diff
@@ -99,3 +101,52 @@ func IsUnsupportedSeriesError(err error) bool {
 	_, ok := err.(*unsupportedSeriesError)
 	return ok
 }
+
+// MaybeCreateVersionFile creates/overwrite charm version file.
+func MaybeCreateVersionFile(path string) error {
+	var charmVersion string
+	var cmdArgs []string
+	var err error
+	// Verify that it is revision control directory.
+	if _, err = os.Stat(filepath.Join(path, ".git")); err == nil {
+		// It is git version control.
+		cmdArgs = []string{"git", "describe", "--dirty"}
+	} else if _, err = os.Stat(filepath.Join(path, ".bzr")); err == nil {
+		// It is baazar.
+		cmdArgs = []string{"bzr", "revision-info"}
+	} else if _, err = os.Stat(filepath.Join(path, ".hg")); err == nil {
+		cmdArgs = []string{"hg", "id", "--id"}
+	} else {
+		logger.Infof("Charm is not in revision control directory")
+		return nil
+	}
+
+	var args []string
+	for pos, arg := range cmdArgs {
+		if pos != 0 {
+			args = append(args, arg)
+		}
+	}
+	cmd := exec.Command(cmdArgs[0], args...)
+	outStr, err := cmd.CombinedOutput()
+	if err != nil {
+		logger.Infof("Command output: %v", outStr)
```

Comment:
This will be displayed to the user - don't log at info level.
If we do log at debug level, it needs more context, eg "charm revision info:"